src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = tail = new Node(null, value, null);
+        } else {
+            Node notTheTailAnyMore = tail;
+            tail = new Node(notTheTailAnyMore, value, null);
+            notTheTailAnyMore.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("given index does not exist");
+        } else {
+            if (index == size) {
+                add(value);
+            } else if (index == 0) {
+                Node notTheHeadAnyMore = head;
+                head = new Node(null, value, notTheHeadAnyMore);
+                notTheHeadAnyMore.prev = head;
+                size++;
+            } else {
+                int currentIndex = 0;
```

Comment:
Don't create repeating code. Here, you are repeating the same logic for finding the node by index in several methods. It would be better to create a separate `findNodeByIndex` method to avoid code duplication.